scripts/west_commands/footprint.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+'''west "footprint" command'''
+
+import argparse
+import os
+import re
+from pathlib import Path
+import subprocess
+import shutil
+
+from west import log
+from runners.core import BuildConfiguration
+from build_helpers import find_build_dir, is_zephyr_build, \
+    FIND_BUILD_DIR_DESCRIPTION
+
+from zcmake import CMakeCache
+from zephyr_ext_common import Forceable
+
+
+class Footprint(Forceable):
+
+    def __init__(self):
+        super(Footprint, self).__init__(
+            'footprint',
+            # Keep this in sync with the string in west-commands.yml.
+            'show footprint information of a binary.',
+            'Show binary code, data, stack usage.',
+            accepts_unknown_args=True)
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=self.description)
+
+        parser.add_argument('-d', '--build-dir',
+                            help=FIND_BUILD_DIR_DESCRIPTION)
+        self.add_force_arg(parser)
+
+        return parser
+
+    def do_run(self, args, runner_args):
+        program = "puncover"
+
+        self.args = args        # for check_force
+        if shutil.which(program) is None:
+            log.die('Unable to find {} program, install and retry.'.format(program))
+
+        # Find the build directory and parse .config and DT.
+        build_dir = find_build_dir(args.build_dir)
+        self.check_force(os.path.isdir(build_dir),
+                         'no such build directory {}'.format(build_dir))
+        self.check_force(is_zephyr_build(build_dir),
+                         "build directory {} doesn't look like a Zephyr build "
+                         'directory'.format(build_dir))
+
+        bcfg = BuildConfiguration(build_dir)
+
+        su_exists = 'CONFIG_STACK_USAGE' in bcfg
+
+
+        self.check_force(su_exists,
+                    'Enable CONFIG_STACK_USAGE in your application to get '
+                    'stack analysis in the report.')
+
+        zephyr_base = Path(os.environ.get('ZEPHYR_BASE',
```

Comment:
`ZEPHYR_BASE` is in the CMake cache in the build directory.
Taking it from there will ensure the same Zephyr base is used, as the one used for the build.

That is safer than taking it from current environment.